frontend/src/pages/search.tsx

```diff
@@ -1,26 +1,71 @@
 import type { GetStaticProps, NextPage } from "next";
-import { useFeatureFlags } from "src/hooks/useFeatureFlags";
+import {
+  Opportunity,
+  SearchFetcher,
+  fetchSearchOpportunities,
+} from "../services/searchfetcher/SearchFetcher";
+import React, { useState } from "react";
 
+import { APISearchFetcher } from "../services/searchfetcher/APISearchFetcher";
+import { MockSearchFetcher } from "../services/searchfetcher/MockSearchFetcher";
+import PageNotFound from "./404";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { useFeatureFlags } from "src/hooks/useFeatureFlags";
 
-import PageNotFound from "./404";
+const useMockData = true;
+const searchFetcher: SearchFetcher = useMockData
+  ? new MockSearchFetcher()
+  : new APISearchFetcher();
 
-const Search: NextPage = () => {
+interface SearchProps {
+  initialOpportunities: Opportunity[];
+}
+
+const Search: NextPage<SearchProps> = ({ initialOpportunities = [] }) => {
   const { featureFlagsManager, mounted } = useFeatureFlags();
+  const [searchResults, setSearchResults] =
+    useState<Opportunity[]>(initialOpportunities);
+
+  const handleButtonClick = () => {
+    performSearch().catch((e) => console.log(e));
+  };
+
+  const performSearch = async () => {
+    const opportunities = await fetchSearchOpportunities(searchFetcher);
+    setSearchResults(opportunities);
+  };
 
   if (!mounted) return null;
   if (!featureFlagsManager.isFeatureEnabled("showSearchV0")) {
     return <PageNotFound />;
   }
 
-  return <>Search Boilerplate</>;
+  return (
+    <>
+      <button onClick={handleButtonClick}>Update Results</button>
+      <ul>
+        {searchResults.map((opportunity) => (
+          <li key={opportunity.id}>
+            {opportunity.id}, {opportunity.title}
+          </li>
+        ))}
+      </ul>
+    </>
+  );
 };
 
-// Change this to GetServerSideProps if you're using server-side rendering
 export const getStaticProps: GetStaticProps = async ({ locale }) => {
```

Comment:
(question): What is the plan for moving to the app router? It looks like we aren't using the app router for the API calls, but still using it for the search page. I believe we could just move the search page to an `app` folder and adopt it for this feature, esp since the API calls aren't using the page router.

Reply:
Newer to nextjs so if you can provide additional context on what this move would do.

I'll take a stab to start the convo:
- I believe we should use `getStaticProps` prerendering for grabbing the initial opportunities - since the initial call should always be the same (we can accomodate the query params later on)

- I think we could use `getServerSideProps` for subsequent api calls after the initial call - so it renders server side on every request 


- I see that the `/pages/api/hello.ts` health check and `subscribe.ts` - are we trying to replicate those? I believe those are proxying calls ?

Reply:
@rylew1 see [Migrate from Next.js "pages" router to newer "app" router](https://github.com/navapbc/template-application-nextjs/issues/66) for more context and discussion. With the app router you can stream data updates from the client to the server. There are some search specific features like [useSearchParams](https://nextjs.org/docs/app/api-reference/functions/use-search-params) which would be useful to take advantage of.  This article: [Adding Search and Pagination
](https://nextjs.org/learn/dashboard-app/adding-search-and-pagination) goes into some detail.

We had discussed updating to the app router with the sendy work, but decided to put if off. The app router can be adopted incrementally, so we could use it for the search page as part of a transition.

Reply:
🤔 if `useSearchParams` is a client component hook, wonder if that works in a server component?

Reply:
I'm going to merge this PR (called 1a), then try to do an `app-router` refactor in a part `1b` PR

in Part 2 we can try to hook up the mock data and actual api cals

Reply:
I'm lacking project context, but sharing some thoughts in case it's helpful:

1. The [`getStaticProps` docs](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props#when-does-getstaticprops-run) are a bit confusing to me, but if you haven't yet, you may want to confirm whether it's only running at build time and whether that's going to be a problem for your use case (i.e if the initial opportunities change after a deploy).
2. The search frontend should likely be a form. Semantically I think that makes sense if the user is entering a search query. If you end up using app router, you [could use Server Actions](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#forms) and [`useFormState`](https://react.dev/reference/react-dom/hooks/useFormState), in place of `getServerSideProps` and the `useState`